### PR TITLE
Cloudwatch implement set_alarm_state

### PR DESCRIPTION
**Pull request by steffyP**
This PR enables the triggering of actions that are set on alarms, when calling `set_alarm_state`.
Limitations: 
 - Currently only SNS is supported as action. 
 - Metrics are not yet evaluated, only the `set_alarm_state` will change the state of an alarm and therefore trigger actions.

### CODE REVIEW

File: localstack/services/cloudwatch/cloudwatch_starter.py [hunk 1]
```diff
@@ -21,6 +28,86 @@ def put_composite_alarm(self):
...
+
+    if alarm.statistic:
+        details["StatisticType"] = "Statistic"
+        details["Statistic"] = alarm.statistic.upper()  # AWS returns uppercase
+    elif alarm.extended_statistic:
+        details["StatisticType"] = "ExtendedStatistic"
+        details["ExtendedStatistic"] = alarm.extended_statistic
+
+    response["Trigger"] = details
+    import json
```
Comment (thrau) [inline localstack/services/cloudwatch/cloudwatch_starter.py:107]: nit: could be a top-level import

**Review by thrau — APPROVED** [review]
fantastic job! i appreciate the effort you put into the test refactoring and making them work against AWS.

File: localstack/services/cloudwatch/cloudwatch_starter.py [hunk 2]
```diff
@@ -21,6 +28,86 @@ def put_composite_alarm(self):
     if not hasattr(cloudwatch_responses.CloudWatchResponse, "put_composite_alarm"):
         cloudwatch_responses.CloudWatchResponse.put_composite_alarm = put_composite_alarm
 
+    @patch(target=FakeAlarm.update_state)
+    def update_state(target, self, reason, reason_data, state_value):
+        old_state = self.state_value
+        target(self, reason, reason_data, state_value)
+
+        # check the state and trigger required actions
+        if self.actions_enabled and old_state != self.state_value:
```
Comment (whummer) [inline localstack/services/cloudwatch/cloudwatch_starter.py:37]: nitpick: we could apply the _Return Early_ pattern here, and check for conditions to return from the function (can help reduce the level of nesting):
```
if not self.actions_enabled or old_state == self.state_value:
    return
...
```

**Review by whummer — APPROVED** [review]
Awesome changes @steffyP ! 🎉 Great to see alarm actions supported now. Thanks for taking the effort to refactor `unittest`->`pytest`. 👍

**Comment by chrisnuval** [comment]
Thank you for supporting this!!

**Pull request merged**